modules/core/src/v2/recovery/initiate.ts

```diff
@@ -0,0 +1,133 @@
+/**
+ * @prettier
+ */
+
+import * as utxolib from '@bitgo/utxo-lib';
+import { BitGo } from '../../bitgo';
+import * as stellar from 'stellar-sdk';
+
+interface ValidateKeyOptions {
+  key: string;
+  source: string;
+  passphrase?: string;
+  isUnsignedSweep: boolean;
+  isKrsRecovery: boolean;
+}
+
+export interface InitiateRecoveryOptions {
+  userKey: string;
+  backupKey: string;
+  bitgoKey?: string; // optional for xrp recoveries
+  recoveryDestination: string;
+  walletPassphrase?: string;
+}
+
+export function getIsKrsRecovery({ backupKey, userKey }: { backupKey: string; userKey: string }): boolean {
+  return backupKey.startsWith('xpub') && !userKey.startsWith('xpub');
+}
+
+export function getIsUnsignedSweep({ backupKey, userKey }: { backupKey: string; userKey: string }): boolean {
+  return backupKey.startsWith('xpub') && userKey.startsWith('xpub');
+}
+
+export function validateKey(
+  bitgo: BitGo,
+  { key, source, passphrase, isUnsignedSweep, isKrsRecovery }: ValidateKeyOptions
+): utxolib.HDNode {
+  if (!key.startsWith('xprv') && !isUnsignedSweep) {
+    // Try to decrypt the key
+    try {
+      if (source === 'user' || (source === 'backup' && !isKrsRecovery)) {
+        return utxolib.HDNode.fromBase58(bitgo.decrypt({ password: passphrase, input: key }));
+      }
+    } catch (e) {
+      throw new Error(`Failed to decrypt ${source} key with passcode - try again!`);
+    }
+  }
+  try {
+    return utxolib.HDNode.fromBase58(key);
+  } catch (e) {
+    throw new Error(`Failed to validate ${source} key - try again!`);
+  }
+}
+
+export function getBip32Keys(
+  bitgo: BitGo,
+  params: InitiateRecoveryOptions,
+  { requireBitGoXpub }: { requireBitGoXpub: boolean }
+): utxolib.HDNode[] {
+  const isKrsRecovery = getIsKrsRecovery(params);
+  const isUnsignedSweep = getIsUnsignedSweep(params);
+  const keys = [
+    // Box A
```

Comment:
Out of curiosity what is Box A/B/C in this context?

Reply:
this refers to the keycard PDF that is generated when you create a wallet